crates/ruff/src/rules/flake8_todos/rules/todos.rs

```diff
@@ -296,25 +239,27 @@ static ISSUE_LINK_REGEX_SET: Lazy<RegexSet> = Lazy::new(|| {
     .unwrap()
 });
 
-pub(crate) fn todos(indexer: &Indexer, locator: &Locator, settings: &Settings) -> Vec<Diagnostic> {
+pub(crate) fn todos(
+    comment_directive_ranges: Vec<(TextRange, usize, TextRange)>,
+    indexer: &Indexer,
+    locator: &Locator,
+    settings: &Settings,
+) -> Vec<Diagnostic> {
```

Comment:
Nit: I wonder if we could avoid the collection of the directive ranges by instead change these rules to be called for every directive:

```rust
todos(
	comment_directive: CommentDirective,
	indexer: &indexer,
	locator: &Locator,
	settings: &Settings
) -> Vec<Diagnostic>
```